pkg/cache/queue/manager.go

```diff
@@ -526,34 +533,49 @@ func (m *Manager) RequeueWorkload(ctx context.Context, info *workload.Info, reas
 	return added
 }
 
-func (m *Manager) assignWorkload(wlInfo *workload.Info, q *LocalQueue) {
-	m.assignedWorkloads[workload.Key(wlInfo.Obj)] = q.Key
-	q.AddOrUpdate(wlInfo)
+// Delete the workload from queue or cluster queue.
+// Does not remove the queue assignment caching.
+func (m *Manager) DeleteWorkload(log logr.Logger, wl *kueue.Workload) {
+	m.Lock()
+	defer m.Unlock()
+	m.deleteWorkload(log, wl)
+	m.DeleteSecondPassWithoutLock(wl)
 }
 
-func (m *Manager) DeleteWorkload(log logr.Logger, wl *kueue.Workload) {
+// Completely forget workload.
+// Delete the workload from queue or cluster queue.
+// Purge queue assignment caching.
+func (m *Manager) DeleteAndForgetWorkload(log logr.Logger, wl *kueue.Workload) {
 	m.Lock()
 	defer m.Unlock()
-	m.deleteWorkloadFromAssignedQueues(log, wl)
+	m.deleteAndForgetWorkload(log, wl)
+}
+
+// Completely forget workload.
+// Delete the workload from queue or cluster queue.
+// Purge queue assignment caching.
+func (m *Manager) deleteAndForgetWorkload(log logr.Logger, wl *kueue.Workload) {
+	m.deleteWorkload(log, wl)
+	delete(m.workloadAssignedQueues, workload.Key(wl))
 	m.DeleteSecondPassWithoutLock(wl)
 }
```

Comment:
I think is it a bit of redundant boilerplate. There's no really a point to divide this logic into public and private functions, if the only function that calls `deleteAndForgetWorkload` is `DeleteAndForgetWorkload`.